File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefLongProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefLongProperty extends CTPrefProperty<Long> implements CTNumberProperty<Long> {
    public CTPrefLongProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, 0L);
    }

    public CTPrefLongProperty(@NotNull CTPreferencesProvider provider, String name, Long def) {
        super(Long.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefLongProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefLongProperty @JvmOverloads constructor(provider: CTPreferencesProvider, name: String, def: Long = 0L) :
    CTPrefProperty<Long>(
        Long::class, provider, name, def
    ), CTNumberProperty<Long>
